### PR TITLE
Add site name from object to display custom domains when present

### PR DESCRIPTION
**Pull request by melvinsoft**
Fixes RED-2428

### CODE REVIEW

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
@melvinsoft this is breaking tests and it's upstream code so I'd rather that we don't change it.

I think a better change would be to override `SITE_NAME` on each SiteConfiguration save on the following order:

```
def save()
    self.site_values['SITE_NAME'] = self.site.domain 
```

This will ensure that at least the site name would be fixed by the next deployment on all sites. I think it's a much needed addition to avoid having us to do this manually and safer than editing upstream code and having to maintain tests for that.

Related code:

https://github.com/appsembler/edx-platform/blob/66cdda525e999bd7f3348d7183e9103a1914caf5/openedx/core/djangoapps/site_configuration/models.py#L78-L100

**Comment by melvinsoft** [comment]
@OmarIthawi Thanks for feedback, change implemented, can you give it another pass?

File: openedx/core/djangoapps/site_configuration/models.py [hunk 1]
```diff
@@ -99,6 +99,10 @@ def save(self, **kwargs):
             scheme=urlsplit(settings.LMS_ROOT_URL).scheme,
             domain=self.site.domain)
 
+        # This ensures if the site has a custom domain set, we set the custom
+        # domain instead the Tahoe URL.
+        self.site_values['SITE_NAME'] = self.site.domain
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/site_configuration/models.py:104]: I think it's worth adding a small test case for it.

**Review by OmarIthawi — APPROVED** [review]
Thanks @melvinsoft!

**Pull request merged**